### ISSUE TITLE
Allow arbitrary queries via web API

### ISSUE THREAD

**Issue by tunnell** [XENON1T/cax#65]
Can we have it so the web API allows arbitrary mongo queries?  This greatly simplifies processing and lightens the load on Mongo if we do it correctly.  For example:

https://gist.github.com/ershockley/0fa384dd4ed36f159cdd754c9aac388f

**Comment by pdeperio** [comment]
is https://github.com/XENON1T/cax/pull/38 supposed to solve this?

**Comment by coderdj** [comment]
Nah, this doesn't exist at all. It's more an API limitation but due to lack of interest (#38 still open) I stopped developing this aspect of the API.

**Comment by ershockley** [comment]
I'm not sure how 'arbitrary' you mean exactly, but so far I haven't had any problem doing complicated mongo queries. Here's an example

In OSG processing we can pass a query like [this](https://github.com/XENON1T/cax/blob/OSG_dev2/cax/tasks/process.py#L198), which is queried via API [here](https://github.com/XENON1T/cax/blob/OSG_dev2/cax/task.py#L137) by calling [this function](https://github.com/XENON1T/cax/blob/OSG_dev2/cax/api.py#L142).
